src/webots/nodes/WbGroup.cpp

```diff
@@ -227,17 +232,25 @@ bool WbGroup::isAValidBoundingObject(bool checkOde, bool warning) const {
 }
 
 void WbGroup::descendantNodeInserted(WbBaseNode *decendant) {
-  if (parentNode()) {
-    WbGroup *pg = dynamic_cast<WbGroup *>(parentNode());
-    WbSlot *ps = dynamic_cast<WbSlot *>(parentNode());
-    if (pg)
-      pg->descendantNodeInserted(decendant);
-    if (ps)
-      emit notifyParentSlot(decendant);
+  if (!parentNode())
+    return;
+
+  WbGroup *pg = dynamic_cast<WbGroup *>(parentNode());
+  if (pg) {
+    pg->descendantNodeInserted(decendant);
+    return;
+  }
+  WbBasicJoint *pj = dynamic_cast<WbBasicJoint *>(parentNode());
+  if (pj) {
```

Comment:
constness:
```suggestion
  const WbBasicJoint *pj = dynamic_cast<WbBasicJoint *>(parentNode());
  if (pj) {
```
Or maybe simpler like this:
```suggestion
  if (dynamic_cast<WbBasicJoint *>(parentNode())) {
```